src/ophys_etl/transforms/classification.py

```diff
@@ -0,0 +1,304 @@
+    )
+
+
+class InferenceParser(ArgSchemaParser):
+    """ Argschema entry point """
+    default_schema = InferenceInputSchema
+    default_output_schema = InferenceOutputSchema
+
+
+def _munge_data(parser: InferenceParser, roi_data: list):
+    """
+    Format the input data for downstream processing.
+    Params
+    ------
+    parser: InferenceParser
+        An instance of InferenceParser
+    roi_data: list
+        List of objects conforming to RoiJsonSchema
+    Returns
+    -------
+    tuple
+        rois (list of coo_matrices), metadata dictionary,
+        traces data (np.1darray) and neuropil traces data (np.1darray)
+    """
+    # Format metadata and multiply for input (all same)
+    metadata = [{
+        "depth": parser.args["depth"],
+        "rig": parser.args["rig"],
+        "targeted_structure": parser.args["targeted_structure"],
+        "full_genotype": parser.args["full_genotype"]
+        }] * len(roi_data)
+    rois = [r["coo_roi"] for r in roi_data]
+    traces = []
+    np_traces = []
+
+    traces_file = h5py.File(parser.args["traces_path"], "r")
+    np_traces_file = h5py.File(parser.args["neuropil_traces_path"], "r")
+    traces_data = traces_file[parser.args["traces_data_key"]]
+    np_traces_data = np_traces_file[parser.args["neuropil_traces_data_key"]]
+    for n in range(len(roi_data)):
+        # Downsample traces by accessing on axis that the h5 file should be
+        # more performant on
+        trace = downsample(
+            traces_data[n, :], parser.args["trace_sampling_fps"],
+            parser.args["downsample_to"])
+        np_trace = downsample(
+            np_traces_data[n, :], parser.args["trace_sampling_fps"],
+            parser.args["downsample_to"])
+        traces.append(trace)
+        np_traces.append(np_trace)
+    traces_file.close()
+    np_traces_file.close()
+    return rois, metadata, traces, np_traces
+
+
+def downsample(trace: np.ndarray, input_fps: int, output_fps: int):
+    """Downsample 1d array using an averaging strategy.
+    If a full bin cannot be populated, the data will be truncated
+    to approximate required output_fps.
+    Example: downsampling from 5 fps to 2 fps
+    for a trace with 13 points would need to drop the last data point.
+
+    Parameters
+    ----------
+    trace: np.ndarray
+        1d array of values to downsample
+    input_fps: int
+        The FPS that the trace data was captured at
+    output_fps: int
+        The desired FPS of the trace
+    Returns
+    -------
+    np.ndarray
+        1d array of values, downsampled to output_fps (may be approximate)
+    """
+    if input_fps == output_fps:
+        return trace
+    points = trace.shape[0]
+    output_size = math.floor(points * output_fps / input_fps)
+    bin_size = math.floor(points / output_size)
+    if points / output_size != bin_size:    # if the bin needed to be rounded
+        warnings.warn(f"Can't evenly downsample {points} points at "
```

Comment:
In this strategy, the user can end up with a different `output_fps` than requested, along with this warning. In [this strategy](https://github.com/AllenInstitute/segmentation-labeling-app/blob/a7175a0ea3bff0062e33d61be68e9b1e14bc97df/slapp/transforms/array_utils.py#L196) the frames per second are as requested. Any particular reason you chose this one?

Reply:
Ah, there it is. That downsampling method will not be correct for the last data point unless the bins are evenly divided. I'd strip the last bin in that case. What do you think?

Reply:
I think losing the last bin is preferable to getting a different FPS than requested

Reply:
Updated to use scipy.signal.resample_poly as discussed offline